src/js/panels.js

```diff
@@ -198,16 +193,377 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
 
+    /*
+     * language panel
+     */
+    fluid.defaults("gpii.firstDiscovery.panel.lang", {
+        gradeNames: ["fluid.prefs.panel", "{that}.options.prefsEditorConnection", "autoInit"],
+        preferenceMap: {
+            "gpii.firstDiscovery.language": {
+                "model.lang": "default",
+                "controlValues.lang": "enum",
+                "stringArrayIndex.lang": "label",
+                "stringArrayIndex.tooltip": "tooltip",
+                "stringArrayIndex.tooltipAtSelect": "tooltipAtSelect"
+            }
+        },
+        components: {
+            attachTooltipOnLang: {
+                type: "gpii.firstDiscovery.panel.lang.attachTooltipOnLang",
+                container: "{lang}.container",
+                options: {
+                    selectors: "{lang}.options.selectors",
+                    modelRelay: {
+                        source: "{lang}.model.lang",
+                        target: "currentSelectedIndex",
+                        singleTransform: {
+                            type: "fluid.transforms.free",
+                            args: {
+                                "langs": "{lang}.options.controlValues.lang",
+                                "currentLang": "{lang}.model.lang"
+                            },
+                            func: "gpii.firstDiscovery.panel.lang.getCurrentSelectedIndex"
+                        }
+                    },
+                    tooltipContentMap: {
+                        "prev": "navButtonTooltip",
+                        "next": "navButtonTooltip",
+                        "langRow": {
+                            tooltip: "{lang}.options.stringArrayIndex.tooltip",
+                            tooltipAtSelect: "{lang}.options.stringArrayIndex.tooltipAtSelect"
+                        },
+                        "langInput": {
+                            tooltip: "{lang}.options.stringArrayIndex.tooltip",
+                            tooltipAtSelect: "{lang}.options.stringArrayIndex.tooltipAtSelect"
+                        }
+                    },
+                    listeners: {
+                        "{lang}.events.afterRender": {
+                            funcName: "{that}.tooltip.applier.change",
+                            args: ["idToContent", {
+                                expander: {
+                                    func: "{that}.tooltip.getTooltipModel"
+                                }
+                            }]
+                        }
+                    }
+                }
+            }
+        },
+        model: {
+            firstLangSelected: false,
+            lastLangSelected: false
+        },
+        modelRelay: [{
+            target: "firstLangSelected",
+            singleTransform: {
+                type: "fluid.transforms.binaryOp",
+                left: "{that}.model.lang",
+                operator: "===",
+                right: {
+                    expander: {
```

Comment:
We should be able to replace this use of an expander with a simple reference - {that}.options.controlValues.lang.0